documentation/error-handling.md

```diff
@@ -0,0 +1,64 @@
+# Error handling
+
+UI extensions support standard web errors. You can add an `unhandledrejection` listener for unhandled promise rejections or an `error` listener for other exceptions like JavaScript runtime errors or resources that fail to load.
+
+```javascript
+// For unhandled promise rejections
+self.addEventListener('unhandledrejection', (error) => {
+  console.warn('event unhandledrejection', error);
+});
+
+// For other exceptions
+self.addEventListener('error', (error) => {
+  console.warn('event error', error);
+});
+```
+
+## Reporting errors
+
+It's possible to use 3rd party libaries like Bugsnag or Sentry for reporting errors. However, it might require extra configuration or setup, depending on which libraries you use, because UI extensions run inside [WebWorkers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API).
+
+### Bugsnag
+
+Follow [Bugsnag's documentation](https://docs.bugsnag.com/platforms/javascript/) to install it into your extension. If you use a [CDN version](https://docs.bugsnag.com/platforms/javascript/cdn-guide/), you need to add polyfill code below as Bugsnag tries to access some variables that are not available in WebWorkers.
+
+```javascript
+/**
+ * CDN polyfill
+ */
+(() => {
+  const document = {
+    documentElement: {
+      outerHTML: '',
+      createElement: () => ({}),
+      clientWidth: 0,
+      clientHeight: 0,
+    },
+    addEventListener: () => {},
+  };
+  const history = {};
+  const window = self;
+  self.window = window;
+  window.document = document;
+  window.history = history;
+})();
+```
+
+There is no need to add custom event listeners as Bugsnag does it for you out of the box.
+
+### Sentry
+
+Follow [Sentry's documentation](https://docs.sentry.io/platforms/javascript/) to install it into your extension. A polyfill is not required for using Sentry. However, Sentry does not add any event listeners out of the box. So, you need to add them manually.
```

Comment:
Sentry also has a React-specific library we can reference: https://docs.sentry.io/platforms/javascript/guides/react/